cross_list_courses/static/cross_list_courses/js/controllers/ListController.js

```diff
@@ -0,0 +1,107 @@
+(function(){
+    /**
+     * Angular controller for the cross listing page.
+     */
+    var app = angular.module('CrossListCourses');
+    app.controller('ListController',
+        ['$scope', '$http', '$timeout', '$document', '$window', '$compile','djangoUrl',
+        function($scope, $http, $timeout, $document, $window, $compile, $djangoUrl){
+
+            $scope.dtInstance = null;
+
+            $scope.formatCourse= function(course_instance) {
+                return course_instance.course.school_id.toUpperCase()+
+                                ' '+course_instance.course.registrar_code+
+                                '-'+course_instance.term.display_name+
+                                '-'+course_instance.course_instance_id;
+
+            };
+
+            $scope.dtOptions = {
+                ajax: function(data, callback, settings) {
+                    var url = $djangoUrl.reverse('icommons_rest_api_proxy',
+                                                   ['api/course/v2/xlist_maps/']);
+
+                    var queryParams = {
+                        offset: data.start,
+                        limit: data.length,
+                        include: 'course_instance',
+                    };
+
+                    $.ajax({
+                        url: url,
+                        method: 'GET',
+                        data: queryParams,
+                        dataSrc: 'data',
+                        dataType: 'json',
+                        success: function(data, textStatus, jqXHR) {
+                            callback({
+                                recordsTotal: data.count,
+                                recordsFiltered: data.count,
+                                data: data.results,
+                            });
+                        },
+                        error: function(data, textStatus, errorThrown) {
+                            $log.error('Error getting data from ' + url + ': '
+                                       + textStatus + ', ' + errorThrown);
+                            callback({
+                                recordsTotal: 0,
+                                recordsFiltered: 0,
+                                data: [],
+                            });
+                        },
+                    });
+                },
+                createdRow: function( row, data, dataIndex ) {
+                    // to use angular directives within the rendered datatable,
+                    // we have to compile those elements ourselves.  joy.
+                    $compile(angular.element(row).contents())($scope);
```

Comment:
Do we need to do this? I don't think we've embedded any angular code in our table rows.